CMakeLists.txt

```diff
@@ -43,7 +43,18 @@ if(WIN32)
   ament_environment_hooks("${BINARY_PATH_HOOK}" "${PYTHONPATH_HOOK}")
 else()
   set(LIBRARY_PATH_HOOK "${AMENT_PACKAGE_TEMPLATE_DIR}/environment_hook/library_path.${SHELL_EXT}")
-  ament_environment_hooks("${BINARY_PATH_HOOK}" "${LIBRARY_PATH_HOOK}" "${PYTHONPATH_HOOK}")
+  # Collect machine architecture triplet for libraries using GNU install dirs
+  execute_process(COMMAND gcc -dumpmachine OUTPUT_VARIABLE MULTIARCH_TRIPLET OUTPUT_STRIP_TRAILING_WHITESPACE)
+  # register multiarch information for .dsv generation
+  if(APPLE)
+    set(LIBRARY_PATH_ENV_VAR "DYLD_LIBRARY_PATH")
+  else()
+    set(LIBRARY_PATH_ENV_VAR "LD_LIBRARY_PATH")
+  endif()
+  set(
+    AMENT_CMAKE_ENVIRONMENT_HOOKS_DESC_multiarch_library_paths
+    "prepend-non-duplicate;${LIBRARY_PATH_ENV_VAR};${MULTIARCH_TRIPLET}")
```

Comment:
@dirk-thomas Is this the right approach (generating a .dsv file)? Or should we instead add a .dsv file explicitly to this repository?

Reply:
The last part needs to contain the relative path from the install prefix. I assume it lacks `lib/` at the moment?

Reply:
Yes, you are correct.  Fixed in 81289b6